tests/test_rois.py

```diff
@@ -0,0 +1,247 @@
+import json
+
+import sqlite3
+import pytest
+import numpy as np
+
+import segmentation_labeling_app.rois.rois as roi_module
+
+
+@pytest.mark.parametrize(("coo_rows, coo_cols, coo_data, video_shape,"
+                          "segmentation_id, roi_id, threshold, expected_mask"), [
+    ([0, 0, 1, 1], [0, 1, 0, 1],
+     [0.75, 0.8, 0.9, 0.85], (3, 3), 1, 1, 0.7,
+     np.array([[1, 1, 0], [1, 1, 0], [0, 0, 0]])),
+    ([0, 0, 0, 1, 1, 1], [0, 1, 2, 0, 1, 2],
+     [0.75, 0.8, 0.5, 0.9, 0.85, 0.4], (3, 3), 1, 1, 0.7,
+     np.array([[1, 1, 0], [1, 1, 0], [0, 0, 0]]))
+])
+def test_binary_mask(coo_rows, coo_cols, coo_data, video_shape,
+                     segmentation_id, roi_id, threshold, expected_mask):
+    test_roi = roi_module.ROI(coo_rows, coo_cols, coo_data,
+                              video_shape, segmentation_id, roi_id)
+    generated_mask = test_roi.generate_binary_mask_from_threshold(threshold=threshold)
+    assert np.array_equal(generated_mask, expected_mask)
+
+
+@pytest.mark.parametrize(("coo_rows, coo_cols, coo_data, video_shape,"
+                          "segmentation_id, roi_id, threshold, expected_edges"),
```

Comment:
Thanks for adding test cases. Please make notes in docstring or comments next to the cases which ones they represent so that it's easier for the reader.

Reply:
Please also update these tests once the coordinate values are returned in the correct order (row,col)